target/linux/mpc85xx/files/arch/powerpc/boot/dts/t30-w.dts

```diff
@@ -0,0 +1,384 @@
+					};
+
+					switch0phy3: switch0phy3@3 {
+						reg = <0x03>;
+						interrupt-parent = <&switch0>;
+					};
+
+					switch0phy4: switch0phy4@4 {
+						reg = <0x04>;
+						interrupt-parent = <&switch0>;
+					};
+				};
+				
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					  
+					port@0 {
+						reg = <0x00>;
+						label = "wan";
+						phy-handle = <&switch0phy0>;
+					};
+
+					port@1 {
+						reg = <1>;
+						label = "lan1";
+						phy-handle = <&switch0phy1>;
+					};
+
+					port@2 {
+						reg = <2>;
+						label = "lan2";
+						phy-handle = <&switch0phy2>;
+					};
+					  
+					port@3 {
+						reg = <3>;
+						label = "lan3";
+						phy-handle = <&switch0phy3>;
+					};
+
+					port@4 {
+						reg = <4>;
+						label = "lan4";
+						phy-handle = <&switch0phy4>;
+					};
+										
+					switch0port6: port@6 {
+						reg = <6>;
+						label = "cpu";
+						ethernet = <&enet0>;
+						phy-mode = "rgmii-id";
+						fixed-link {
+							speed = <1000>;
+							full-duplex;
+						};
+					};
+				};
+			};
+		};
+		
+		mdio@25000 {
+			compatible = "fsl,etsec2-tbi";
+			reg = <0x25000 0x1000 0xb1030 0x4>;
+			tbi1: tbi-phy@16 {
+				reg = <0x10>;
+			};
+		};
+		
+		mdio@26000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "fsl,etsec2-tbi";
+			reg = <0x26000 0x1000 0xb1030 0x4>;
+		};
+
+		enet0: ethernet@b0000 {
+			compatible = "fsl,etsec2";
+			phy-connection-type = "rgmii-id";
+			fixed-link {
+				speed = <1000>;
+				full-duplex;
+			};
+		};
+
+		enet1: ethernet@b1000 {
+			status = "disabled";
+		};
+
+		enet2: ethernet@b2000 {
+			status = "disabled";
+		};
+
+		gpio0: gpio@f000 {
+			compatible = "fsl,mpc8572-gpio", "fsl,pq3-gpio";
+			reg = <0xf000 0x1000>;
+			interrupts = <47 2>;
+			interrupt-parent = <&mpic>;
+			#gpio-cells = <2>;
+			gpio-controller;
+		};
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+
+		attn_orange: attn_orange {
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			label = "orange:attn";
+		};
+		
+		status_red: status_red {
+			gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
+			label = "red:status";
+		};
+		
+		mode_green: mode_green {
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+			label = "green:mode";
+		};
+		
+		wap_green: wap_green {
+			gpios = <&gpio0 5 GPIO_ACTIVE_HIGH>;
+			label = "green:wap";
+			default-state = "keep";
+		};
+		
+		wap_orange: wap_orange {
+			gpios = <&gpio0 6 GPIO_ACTIVE_HIGH>;
+			label = "orange:wap";
+			default-state = "keep";
+		};
+		
+		failover_green: failover_green {
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+			label = "green:failover";
+		};
+	};
+
+	pci0: pcie@ffe09000 {
+		cell-index = <0x01>;
+		compatible = "fsl,mpc8548-pcie";
+		device_type = "pci";
+		#interrupt-cells = <0x01>;
+		#size-cells = <0x02>;
+		#address-cells = <0x03>;
+		reg = <0x00 0xffe09000 0x00 0x1000>;
+		bus-range = <0x00 0xff>;
+		ranges = <0x2000000 0x0 0xa0000000 0x0 0xa0000000 0x0 0x20000000 
+			0x1000000 0x00 0x00 0x00 0xffc10000 0x00 0x10000>;
```

Comment:
Cleaning up huge ranges, reg, or interrupt-maps array is tricky. Especially if you don't have the board or a similar device. I do think this is supposed to contain two entries like this:

    ranges = <0x2000000 0x0 0xa0000000 0x0 0xa0000000 0x0 0x2000000>,
             <0x1000000 0x0 0x0        0x0 0xffc10000 0x0 0x10000>;

Reply:
I don't mind formatting it that way, but functionally it should be the same right ?

Reply:
Yes, this is confusing (they work sort-of-like c-arrays). But there's something written about this in
https://elinux.org/Device_Tree_Mysteries Example 3.1: 

```
Example 3.1

Two tuples, one element per tuple:

   interrupts = <17>, <4>;

One tuple, two elements per tuple:

   interrupts = <17 4>;

Three tuples, with formats of:
   1) phandle, interrupt
   2) phandle, interrupt, flags
   2) phandle, interrupt

   interrupts-extended = < &INTC_1 17 >, < &INTC_2 23 8 >, < &INTC_4 27 >;


There are no type declarations for arrays and tuples, so the
dtc compiler can not enforce conformance with tuple size in the value
declarations.

Thus the above three tuples would properly compile if specified as:

   interrupts-extended = < &INTC_1 17 &INTC_2 >, < 23 8 >, < &INTC_4 27 >;

even though it appears not to match the formats of the three tuples.
If you compile a device tree source file containing each of the two above
definitions of interrupts, then decompile the resulting .dtb file, you will
see the dtc simply converts both forms to be:

   <0x2 0x11 0x3 0x17 0x8 0x4 0x1b>

(In the test .dts file I created, &INTC_1 has the value 0x2, &INTC_2 has
the value 0x3, and &INTC_4 has the value 0x4.)
```

Reply:
It sounds like a cosmetic thing, how can I confirm what is should look like ?